File: TDS_Server/Dto/Map/Creator/MapCreatorPosition.cs
using System;
using System.Collections.Generic;
using System.Text;
using TDS_Common.Dto.Map;
using TDS_Common.Enum;

namespace TDS_Server.Dto.Map.Creator
{
    public class MapCreatorPosition
    {
        public int Id { get; set; }
        public EMapCreatorPositionType Type { get; set; }
        public object Info { get; set; } = 0;
        public float PosX { get; set; }
        public float PosY { get; set; }
        public float PosZ { get; set; }
        public float RotX { get; set; }
        public float RotY { get; set; }
        public float RotZ { get; set; }

        public MapCreatorPosition() { }

        public MapCreatorPosition(int id, Position3DDto pos)
        {
            Id = id;
            PosX = pos.X;
            PosY = pos.Y;
            PosZ = pos.Z;
        }

        public MapCreatorPosition(int id, Position4DDto pos)
        {
            Id = id;
            PosX = pos.X;
            PosY = pos.Y;
            PosZ = pos.Z;
            RotZ = pos.Rotation;
        }
    }
}

File: TDS_Client/Window/angular/projects/main/src/app/components/mapcreator/models/mapCreatorPosition.ts
import { MapCreatorPositionType } from '../enums/mapcreatorpositiontype.enum';

export class MapCreatorPosition {
    /** Id */
    [0]: number;
    /** Type */
    [1]: MapCreatorPositionType;
    /** Info */
    [2]?: string | number;
    /** PosX */
    [3]: number;
    /** PosY */
    [4]: number;
    /** PosZ */
    [5]: number;
    /** RotX */
    [6]: number;
    /** RotY */
    [7]: number;
    /** RotZ */
    [8]: number;

    constructor(id: number, type: MapCreatorPositionType, posX: number, posY: number, posZ: number, rotX: number, rotY: number, rotZ: number) {
        this[0] = id;
        this[1] = type;
        this[3] = posX;
        this[4] = posY;
        this[5] = posZ;
        this[6] = rotX;
        this[7] = rotY;
        this[8] = rotZ;
    }
}
